pkg/utils/gwlog/metadata.go

```diff
@@ -0,0 +1,86 @@
+package gwlog
+
+import (
+	"context"
+	"github.com/google/uuid"
+)
+
+type key string
+
+const metadata key = "metadata"
+
+const traceID string = "trace_id"
+
+type metadataValue struct {
+	m map[string]string
+}
+
+func (mv *metadataValue) get(key string) string {
+	return mv.m[key]
+}
+
+func (mv *metadataValue) set(key, val string) {
+	mv.m[key] = val
+}
+
+func newMetadata() *metadataValue {
+	return &metadataValue{
+		m: make(map[string]string),
+	}
+}
+
+func NewTrace(ctx context.Context) context.Context {
+	currID := uuid.New()
+
+	newCtx := context.WithValue(ctx, metadata, newMetadata())
+	AddMetadata(newCtx, traceID, currID.String())
+
+	return newCtx
+}
+
+func AddMetadata(ctx context.Context, key, value string) {
+	if ctx.Value(metadata) != nil {
+		ctx.Value(metadata).(*metadataValue).set(key, value)
+	}
+}
+
+func GetMetadata(ctx context.Context) []interface{} {
```

Comment:
I was flip flopping around keeping this. My reasoning to keep it around is to allow the developer to choose to print out the metadata at any time instead of just at the end of the trace. But maybe I can keep that option open for the future and whoever needs it can make this a public method.

I think it still might be a nice helper, but maybe can be private.